lib/paperclip/storage/eitheror.rb

```diff
@@ -73,7 +73,7 @@ def usable_storage
         either_exists = @either.exists?
         or_exists = @or.exists?
 
-        return @either if either_exists || !or_exists
+        return @either if !or_exists || either_exists
```

Comment:
I'm not sure it makes any difference since the calculations are mare upfront. 
If you inline those variables, then I think it would probably be faster in the scenario we currently have.